File: conversions/strip-ansi/elide-strip-ansi.py
#!/usr/bin/env python3
"""
Python Integration Example for elide-strip-ansi
"""

# from elide import require
# stripAnsi = require('./elide-strip-ansi.ts')

print("=== Python Calling TypeScript Strip ANSI ===\n")

# Example: Clean log files
# log = "\x1b[32mINFO\x1b[0m Server started"
# clean = stripAnsi.strip(log)
# print(clean)  # "INFO Server started"

# Example: Process terminal output
# def clean_logs(log_lines):
#     return [stripAnsi.strip(line) for line in log_lines]

print("Real-world use case:")
print("- Python log processor needs to clean ANSI codes")
print("- Uses same TypeScript implementation as Node.js")
print("- Consistent text cleaning across services")
print()

print("When Elide Python API is ready:")
print("  from elide import require")
print("  stripAnsi = require('./elide-strip-ansi.ts')")
print("  clean = stripAnsi.strip(colored_text)")

File: conversions/strip-ansi/elide-strip-ansi.rb
#!/usr/bin/env ruby

# Ruby Integration Example for elide-strip-ansi

puts "=== Ruby Calling TypeScript Strip ANSI ===\n"

# stripAnsi = Elide.require('./elide-strip-ansi.ts')
# log = "\x1b[31mERROR\x1b[0m Connection failed"
# clean = stripAnsi.strip(log)
# puts clean  # "ERROR Connection failed"

puts "Real-world use case:"
puts "- Ruby worker processes colored terminal output"
puts "- Uses same implementation as Node.js logger"
puts "- Consistent log cleaning across services"
puts ""

puts "Ruby Integration (when ready):"
puts "  stripAnsi = Elide.require('./elide-strip-ansi.ts')"
puts "  clean_text = stripAnsi.strip(colored_text)"
